ancestrymanager/ancestrymanager_test.go

```diff
@@ -414,6 +414,225 @@ func TestGetAncestors(t *testing.T) {
+				Type: "cloudresourcemanager.googleapis.com/Project",
+			},
+			v3Responses: map[string]*crmv3.Project{},
+			v1Responses: map[string][]*crmv1.Ancestor{
+				"foo": {
+					{ResourceId: &crmv1.ResourceId{Id: "foo", Type: "project"}},
+					{ResourceId: &crmv1.ResourceId{Id: "bar", Type: "folder"}},
+					{ResourceId: &crmv1.ResourceId{Id: "qux", Type: "organization"}},
+				},
+			},
+			want:    []string{"projects/foo", "folders/bar", "organizations/qux"},
+			parent:  "//cloudresourcemanager.googleapis.com/folders/bar",
+			v1Count: 1,
+		},
+		{
+			name: "folder changed from folders/bar1 to folders/bar2",
+			data: tfdata.NewFakeResourceData(
+				"google_project",
+				p.ResourcesMap["google_project"].Schema,
+				map[string]interface{}{
+					"project_id": "foo",
+					"folder_id":  "folders/bar2",
+				},
+			),
+			asset: &resources.Asset{
+				Type: "cloudresourcemanager.googleapis.com/Project",
+			},
+			v3Responses: map[string]*crmv3.Project{
+				"folders/bar2":       {Name: "folders/bar2", Parent: "organizations/qux2"},
+				"organizations/qux2": {Name: "organizations/qux2", Parent: ""},
+			},
+			v1Responses: map[string][]*crmv1.Ancestor{
+				"foo": {
+					{ResourceId: &crmv1.ResourceId{Id: "foo", Type: "project"}},
+					{ResourceId: &crmv1.ResourceId{Id: "bar", Type: "folder"}},
+					{ResourceId: &crmv1.ResourceId{Id: "qux", Type: "organization"}},
+				},
+			},
+			want:    []string{"projects/foo", "folders/bar2", "organizations/qux2"},
+			parent:  "//cloudresourcemanager.googleapis.com/folders/bar2",
+			v3Count: 1,
+			v1Count: 1,
+		},
+		{
+			name: "folder changed from folders/bar1/folders/bar2 to folders/bar2",
+			data: tfdata.NewFakeResourceData(
+				"google_project",
+				p.ResourcesMap["google_project"].Schema,
+				map[string]interface{}{
+					"project_id": "foo",
+					"folder_id":  "folders/bar2",
+				},
+			),
+			asset: &resources.Asset{
+				Type: "cloudresourcemanager.googleapis.com/Project",
+			},
+			v3Responses: map[string]*crmv3.Project{},
+			v1Responses: map[string][]*crmv1.Ancestor{
+				"foo": {
+					{ResourceId: &crmv1.ResourceId{Id: "foo", Type: "project"}},
+					{ResourceId: &crmv1.ResourceId{Id: "bar", Type: "folder"}},
+					{ResourceId: &crmv1.ResourceId{Id: "bar2", Type: "folder"}},
+					{ResourceId: &crmv1.ResourceId{Id: "qux", Type: "organization"}},
+				},
+			},
+			want:    []string{"projects/foo", "folders/bar2", "organizations/qux"},
+			parent:  "//cloudresourcemanager.googleapis.com/folders/bar2",
+			v1Count: 1,
+		},
+		{
+			name: "folder changed from folders/bar2 to folders/bar1/folders/bar2",
+			data: tfdata.NewFakeResourceData(
+				"google_project",
+				p.ResourcesMap["google_project"].Schema,
+				map[string]interface{}{
+					"project_id": "foo",
+					"folder_id":  "folders/bar2",
+				},
+			),
+			asset: &resources.Asset{
+				Type: "cloudresourcemanager.googleapis.com/Project",
+			},
+			v3Responses: map[string]*crmv3.Project{},
```

Comment:
```suggestion
			v3Responses: map[string]*crmv3.Project{
				"folders/bar2":       {Name: "folders/bar", Parent: "folders/bar2"},
			},
```

Reply:
are we still filling the v3 responses even though they are not accessed?

Reply:
In this test (moving from parent -> child) I would expect the flow to be:

1. Resource starts at `organizations/qux/folders/bar2/projects/foo`
2. Resource folder_id is set to `folders/bar` (a child of `folders/bar2`)
3. ancestryManager pre-warms the cache with GetAncestry, which fetches the old ancestry: `organizations/qux/folders/bar2/projects/foo`
4. Because the old folder doesn't match the new one, the ancestryManager falls back to the v3 API to get the ancestry for `folders/bar`
5. After getting the parent for the child folder, it ends up back at `folders/bar2` and uses the cache for the rest.

Does that match your expectations?